src/etc/inc/certs.inc

```diff
@@ -269,6 +269,48 @@ function cert_create(&$cert, $caref, $keylen, $lifetime, $dn, $digest_alg = 'sha
     return true;
 }
 
+function sign_cert_for_csr(&$cert, $caref, $csr, $lifetime, $digest_alg, $x509_extensions = 'usr_cert')
+{
+    $ca = &lookup_ca($caref);
+    if (!$ca) {
+        return false;
+    }
+
+    $ca_str_crt = base64_decode($ca['crt']);
+    $ca_str_key = base64_decode($ca['prv']);
+    $ca_res_crt = openssl_x509_read($ca_str_crt);
+    $ca_res_key = openssl_pkey_get_private(array(0 => $ca_str_key, 1 => ""));
+    if (!$ca_res_key) {
+        return false;
+    }
+    $ca_serial = ++$ca['serial'];
+
+    $args = array(
+        'private_key_type' => OPENSSL_KEYTYPE_RSA,
```

Comment:
This may cause trouble if the key type does not match. RSA is not the only option.

Reply:
Exactly. According to http://php.net/manual/en/function.openssl-csr-new.php, these are the other types:
OPENSSL_KEYTYPE_DSA
OPENSSL_KEYTYPE_DH
OPENSSL_KEYTYPE_EC

In real-world scenarios, I'd believe at least OPENSSL_KEYTYPE_EC would be relevant for supporting elliptic curve cryptography certficates.